### ISSUE TITLE
posit

### ISSUE THREAD

**Issue by timm** [researchart/rose6icse#18]
https://github.com/researchart/rose6icse/tree/master/submissions/available/posit 

Contact:
Profir-Petru Partachi:University College London:PPPI
Dr Earl T. Barr:University College London:barr
Dr. Santanu Kumar Dash:University of Surrey:askdash
Dr. Christoph Treude:University of Adelaide:ctreude

# seeking   Available badge

**Comment by dmz-peerreview** [comment]
Dear authors, 

Thanks for your submission. Overall, I have to raise the following concerns: 
* The versions of the packages required and dependencies seem to be not correctly specified
* An additional package not mentioned in the documentation is installed via your `setup.py` (`tqdm`)
* `pip3 install .` using `setup.py` currently fails with: 
```
Processing /POSIT
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-_4biuxqx-build/setup.py", line 26, in <module>
        'scikit-learn', 'tqdm'
      File "/usr/lib/python3.6/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 372, in __init__
        _Distribution.__init__(self, attrs)
      File "/usr/lib/python3.6/distutils/dist.py", line 253, in __init__
        getattr(self.metadata, "set_" + key)(val)
      File "/usr/lib/python3.6/distutils/dist.py", line 1207, in set_requires
        distutils.versionpredicate.VersionPredicate(v)
      File "/usr/lib/python3.6/distutils/versionpredicate.py", line 114, in __init__
        raise ValueError("expected parenthesized list: %r" % paren)
    ValueError: expected parenthesized list: '-learn'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-_4biuxqx-build/
```

Please check (maybe when the 2nd review comes it). As it stands, the package seems to be currently not functional.

**Comment by PPPI** [comment]
Dear reviewer,

Thank you for spotting this. 

TQDM was used in older code for baselines not in the paper, but since it is still in the published code, it should have been included in the requrements for the SVM baseline. This has been updated in the repository README.

WRT to the setup.py, this was not the intended way to run the package, and the file was autogenerated. This has now been fixed by patching `requires` to `install_requires` in the file to allow PyPI package names. However, I still recommend running the code as per the README descriptions.

Sincerely,
Profir-Petru Partachi

**Comment by timm** [comment]
@PPPI
@barr
@askdash
@ctreude

we are stalled on this review due to a pip3 install issue. can you reproduce the above bug? please advise (and quickly).

**Comment by PPPI** [comment]
Dear all,

The bug should have been fixed in the latest revision, I have removed a previous comment waiting on review number 2.

The TL;DR: 
cause: the setup.py had a PyPI name rather than a package name for `scikit-learn`.
solution: change `scikit-learn` to `sklearn`
or
solution: git pull/clone latest revision

I would like to still emphasize that this package was not made to be run after a pip install, rather in the cloned directory as per the README.md

If there are other issues install issues or concerns, feel free to ping me (I am the corresponding author).

Sincerely,
Profir-Petru Partachi

Edit:
For clarity, I will include the setup.py link [here](https://github.com/PPPI/POSIT/blob/master/setup.py).
The change is on line '26', mind that line '27' contains the missed by us dependency for the SVM baseline TQDM, which is in the repo readme but not the submission as of 17.02.2020.

**Comment by minkull** [comment]
@JonDoe-icse20Artifacts look forward to your review

**Comment by dmz-peerreview** [comment]
Dear authors,
Great work, thank you! With that version, I was able to run the scripts and recommend providing the badge (provided my co-reviewer raises no further issues)

**Comment by JonDoe-ArtifactsTrack** [comment]
Dear All,
thanks for your submission. Indeed, following to your followup from @PPPI i was also able to get the artifact set and running. So thanks to co-reviewer @dmz-peerreview for also highlighting the flaws. 

I agree that at this point the artifact can be attributed the Available (i.e. functional + placed in accessible archival repo) badge.

**Comment by PPPI** [comment]
Dear all,

Thank you for helping us improve the repository and making it properly available with all details surfaced correctly.

Kind regards,
Profir-Petru Partachi
